partiql-ast/src/experimental/ast.rs

```diff
@@ -588,7 +588,7 @@ pub struct Call {
 #[derive(Clone, Debug, PartialEq)]
 pub struct CallAgg {
     pub func_name: SymbolPrimitive,
-    args: Vec<Box<Expr>>,
+    args: Vec<Expr>,
```

Comment:
Here the `Box` was removed from the `Vec`, but in `CreateIndex`, a `Box` was added inside the `Vec`.

Is there a reason for the difference? Should we standardize all to be the same? I know general recommendations are that `Box` aren't necessary in `Vec`, it may be worth always doing it as `Expr` is large. Perhaps we'll need to add `#[allow(clippy::vec_box)]` to the members we do this for...

Reply:
Good call-out.

I agree, we should standardize; it seems like the main reason is that _args_ is private in which case clippy prompts the following:

```bash
warning: `Vec<T>` is already on the heap, the boxing is unnecessary
```

When marked as _pub_ the warning goes away.

```bash
$ cargo clippy -V
clippy 0.1.58 (db9d1b20 2022-01-20)
```

I'll look into understanding the reason further; sending a new commit for now!